src/main_nep/nep4.cu

```diff
@@ -174,22 +173,20 @@ NEP4::NEP4(char* input_dir, Parameters& para, int N, int N_times_max_NN_angular)
   nep_data.dq_dz.resize(N_times_max_NN_angular * ann.dim);
   nep_data.q.resize(N * ann.dim);
   nep_data.gnn_descriptors.resize(N * ann.dim);
-  nep_data.gnn_messages.resize(N * ann.dim);
   nep_data.dU_dq.resize(N * ann.dim);
   nep_data.s.resize(N * (nep_para.n_max_angular + 1) * NUM_OF_ABC);
-  nep_data.parameters.resize(ann.num_para + gnn.num_para);
+  nep_data.parameters.resize(ann.num_para);
 }
 
-void NEP4::update_potential(const float* parameters, ANN& ann, GNN& gnn)
+void NEP4::update_potential(const float* parameters, ANN& ann, Para& nep_para)
 {
-  // ann
   ann.w0 = parameters;
   ann.b0 = ann.w0 + ann.num_neurons1 * ann.dim;
   ann.w1 = ann.b0 + ann.num_neurons1;
   ann.b1 = ann.w1 + ann.num_neurons1;
   ann.c = ann.b1 + 1;
-  // gnn
-  gnn.theta = parameters + ann.num_para;
+  ann.d =
+    ann.c + nep_para.num_types_sq * (nep_para.n_max_angular + 1) * (nep_para.basis_size_radial + 1);
```

Comment:
Have you re-added the radial descriptors, or is this only with angular components?

Reply:
Here the name "basis_size_radial" is historical and misleading. It is actually the "radial function basis size for angular descriptors". 

There are still angular descriptor only in NEP4. I am considering adding the radial part.